test/lib/custom_domain/custom_domain_langs_test.rb

```diff
@@ -0,0 +1,86 @@
+      assert_nil(@custom_domain_langs.custom_domain_lang_by_url('http://otherdomain.com/dir/path/test.html'))
+
+      assert_equal('fr', lang_for(@custom_domain_langs.custom_domain_lang_by_url('http://foo.com')))
+      assert_equal('fr', lang_for(@custom_domain_langs.custom_domain_lang_by_url('http://foo.com/')))
+      assert_equal('fr', lang_for(@custom_domain_langs.custom_domain_lang_by_url('http://foo.com/test.html')))
+
+      assert_equal('ja', lang_for(@custom_domain_langs.custom_domain_lang_by_url('http://foo.com/path')))
+      assert_equal('ja', lang_for(@custom_domain_langs.custom_domain_lang_by_url('http://foo.com/path/')))
+      assert_equal('ja', lang_for(@custom_domain_langs.custom_domain_lang_by_url('http://foo.com/path/dir')))
+      assert_equal('ja', lang_for(@custom_domain_langs.custom_domain_lang_by_url('http://foo.com/path/test.html')))
+
+      assert_equal('zh-CHS', lang_for(@custom_domain_langs.custom_domain_lang_by_url('http://foo.com/dir/path')))
+      assert_equal('zh-CHS', lang_for(@custom_domain_langs.custom_domain_lang_by_url('http://foo.com/dir/path/')))
+      assert_equal('zh-CHS', lang_for(@custom_domain_langs.custom_domain_lang_by_url('http://foo.com/dir/path/dir')))
+      assert_equal('zh-CHS', lang_for(@custom_domain_langs.custom_domain_lang_by_url('http://foo.com/dir/path/test.html')))
+
+      assert_equal('en', lang_for(@custom_domain_langs.custom_domain_lang_by_url('http://english.foo.com/dir/path')))
+      assert_equal('en', lang_for(@custom_domain_langs.custom_domain_lang_by_url('http://english.foo.com/dir/path/')))
+      assert_equal('en', lang_for(@custom_domain_langs.custom_domain_lang_by_url('http://english.foo.com/dir/path/test.html')))
+    end
+
+    def test_get_custom_domain_lang_by_url_with_nested_paths
+      custom_domain_langs_setting = {
+        'ja' => { 'url' => 'foo.com/path' },
+        'en' => { 'url' => 'foo.com/path/en' },
+        'fr' => { 'url' => 'foo.com/path/fr' }
+      }
+      custom_domain_langs = CustomDomainLangs.new(custom_domain_langs_setting)
+      assert_equal('ja', lang_for(custom_domain_langs.custom_domain_lang_by_url('http://foo.com/path')))
+      assert_equal('en', lang_for(custom_domain_langs.custom_domain_lang_by_url('http://foo.com/path/en')))
+      assert_equal('fr', lang_for(custom_domain_langs.custom_domain_lang_by_url('http://foo.com/path/fr')))
+    end
+
+    def test_to_html_swapper_hash
+      expected = {
+        'foo.com' => 'fr',
+        'foo.com/path' => 'ja',
+        'foo.com/dir/path' => 'zh-CHS',
+        'english.foo.com' => 'en'
+      }
+
+      assert(hash_equals(expected, @custom_domain_langs.to_html_swapper_hash))
+    end
+
+    private
+
+    def lang_for(custom_domain_lang)
+      custom_domain_lang.lang
+    end
+
+    def host_and_path_without_trailing_slash_for(custom_domain_lang)
+      custom_domain_lang.host_and_path_without_trailing_slash
+    end
```

Comment:
i'd normally say to remove these if it wasn't test code since it seems redundant. but I suppose it brings it to the left more.

Reply:
Actually it's not being used, though it's supposed to be used for source custom domain. Removed in 6af38847360df0226644a99e8a990e1b2dc48297.